bitnami/cassandra/templates/headless-svc.yaml

```diff
@@ -1,9 +1,20 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ include "cassandra.fullname" . }}-headless
-  labels: {{- include "cassandra.labels" . | nindent 4 }}
-  annotations: {{ include "cassandra.tplValue" ( dict "value" .Values.service.annotations "context" $) | nindent 4 }}
+  name: {{ include "common.names.fullname" . }}-headless
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if or .Values.service.annotations .Values.commonAnnotations }}
```

Comment:
I don't think it is necessary to add an `if`. The worst case scenario would be an empty `annotations` section

Reply:
Correct. But for users doing `helm template ... && git commit ... && kubectl apply ...` to follow GitOps, it makes a difference IMHO.
I mean, both manifests below are valid YAMLs and they have the same effect:
- A:
```yaml
# helm template cassandra bitnami/cassandra -s templates/headless-svc.yaml---
# Source: cassandra/templates/headless-svc.yaml
apiVersion: v1
kind: Service
metadata:
  name: cassandra-headless
  labels:
    app.kubernetes.io/name: cassandra
    helm.sh/chart: cassandra-6.0.0
    app.kubernetes.io/instance: cassandra
    app.kubernetes.io/managed-by: Helm
spec:
...
```
- B:
```yaml
# helm template cassandra bitnami/cassandra -s templates/headless-svc.yaml---
# Source: cassandra/templates/headless-svc.yaml
apiVersion: v1
kind: Service
metadata:
  name: cassandra-headless
  labels:
    app.kubernetes.io/name: cassandra
    helm.sh/chart: cassandra-6.0.0
    app.kubernetes.io/instance: cassandra
    app.kubernetes.io/managed-by: Helm
  annotations:
spec:
...
```
But one of them includes an unnecessary field which adds no values

Reply:
Yes, but it comes at the cost of having more complex templates with more ifs with `and` and `or` operators. For now don't change anything. I don't want to block this